src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1289,4 +1290,1373 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+                        ShowContinueError(state, "For heating objects, the companion must be a cooling object, and vice-versa");
+                        ShowFatalError(state, "Invalid companion object causes program termination");
+                    }
+                    thisHP.companionHeatPumpCoil = &potentialCompanionCoil;
+                    break;
+                }
+            }
+            if (!thisHP.companionHeatPumpCoil) {
+                ShowSevereError(state, "Could not find matching companion heat pump coil.");
+                ShowContinueError(state, "Base coil: " + thisCoilName);
+                ShowContinueError(state, "Looking for companion coil named: " + targetCompanionName);
+                ShowFatalError(state, "Simulation aborts due to previous severe error");
+            }
+        }
+    }
+}
+
+void EIRFuelFiredHeatPump::processInputForEIRPLHP(EnergyPlusData &state)
+{
+    struct ClassType
+    {
+        DataPlant::PlantEquipmentType thisType;
+        std::string nodesType;
+        std::function<Real64(Real64, Real64)> calcLoadOutletTemp;
+        std::function<Real64(Real64, Real64)> calcQsource;
+        std::function<Real64(Real64, Real64)> calcSourceOutletTemp;
+
+        ClassType(DataPlant::PlantEquipmentType _thisType,
+                  std::string _nodesType,
+                  std::function<Real64(Real64, Real64)> _tLoadOutFunc,
+                  std::function<Real64(Real64, Real64)> _qSrcFunc,
+                  std::function<Real64(Real64, Real64)> _tSrcOutFunc)
+            : thisType(_thisType), nodesType(std::move(_nodesType)), calcLoadOutletTemp(std::move(_tLoadOutFunc)), calcQsource(std::move(_qSrcFunc)),
+              calcSourceOutletTemp(std::move(_tSrcOutFunc))
+        {
+        }
+    };
+    std::vector<ClassType> classesToInput = {
+        ClassType{DataPlant::PlantEquipmentType::HeatPumpFuelFiredCooling,
+                  "Chilled Water Nodes",
+                  EIRPlantLoopHeatPumps::EIRFuelFiredHeatPump::subtract,
+                  EIRPlantLoopHeatPumps::EIRFuelFiredHeatPump::add,
+                  EIRPlantLoopHeatPumps::EIRFuelFiredHeatPump::add},
+        ClassType{DataPlant::PlantEquipmentType::HeatPumpFuelFiredHeating,
+                  "Hot Water Nodes",
+                  EIRPlantLoopHeatPumps::EIRFuelFiredHeatPump::add,
+                  EIRPlantLoopHeatPumps::EIRFuelFiredHeatPump::subtract,
+                  EIRPlantLoopHeatPumps::EIRFuelFiredHeatPump::subtract},
+    };
+
+    bool errorsFound = false;
+    auto &cCurrentModuleObject = state.dataIPShortCut->cCurrentModuleObject;
+    for (auto &classToInput : classesToInput) {
+        cCurrentModuleObject = DataPlant::PlantEquipTypeNames[static_cast<int>(classToInput.thisType)];
+
+        auto objType = (DataLoopNode::ConnectionObjectType)getEnumerationValue(BranchNodeConnections::ConnectionObjectTypeNamesUC,
```

Comment:
No `auto` for scalars.

Reply:
Switched to `std::string &`.

Reply:
Ummm ... the type is `DataLoopNode::ConnectionObjectType`. Also you should use C++-style `static_cast<X>` (which the compiler checks) as opposed to C-style `(X)` which the compiler doesn't check.

Reply:
Addressed in previous changes---auto was changed to `std::string &`; and the C++ static_cast was used.